x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/left/tabs/visualize_tab.tsx

```diff
@@ -135,6 +138,12 @@ export const VisualizeTab = memo(() => {
 
   if (hasGraphRepresentation && isGraphFeatureEnabled) {
     options.push(graphVisualizationButton);
+
+    if (activeVisualizationId === GRAPH_ID) {
```

Comment:
similar to the other comment, if you only want it to fire once, consider moving the call inside `onChangeCompressed`

Reply:
thanks, I changed it to use onChange